python/kvikio/_lib/libnvcomp.pyx

```diff
@@ -57,305 +63,184 @@ class pyNvcompType_t(Enum):
     pyNVCOMP_TYPE_BITS = nvcompType_t.NVCOMP_TYPE_BITS
 
 
-# _Cascaded Compressor / Decompressor
-cdef class _CascadedCompressor:
-    cdef __CascadedCompressor* c
+cdef class _nvcompManager:
+    # Temporary storage for factory allocated manager to prevent cleanup
+    cdef shared_ptr[nvcompManagerBase] _mgr
+    cdef nvcompManagerBase* _impl
+    cdef shared_ptr[CompressionConfig] _compression_config
+    cdef shared_ptr[DecompressionConfig] _decompression_config
```

Comment:
```suggestion
    cdef shared_ptr[DecompressionConfig] _decompression_config
    # Other internal bits to keep objects alive
    cdef Array _scratch_buffer
```

Reply:
The scratch buffer is owned by the caller, `nvcomp.py`, I don't think it needs to be held here?